### PR TITLE
Document JDBC API

### LINKED ISSUE #316: Add API -> JDBC section

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

Document JDBC API

image

## 🔨 Related Issues

Closes https://github.com/spiceai/docs/issues/316

### CODE REVIEW

File: spiceaidocs/docs/api/jdbc/index.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+sidebar_label: 'JDBC'
+sidebar_position: 2
+description: 'JDBC API Documentation'
+pagination_prev: null
+pagination_next: null
+---
+
+[JDBC](https://docs.oracle.com/javase/tutorial/jdbc/basics/index.html) (Java Database Connectivity) is a standard API for connecting to and interacting with databases.
+
+Spice supports JDBC clients through a JDBC driver implementation based on the [Flight SQL](https://arrow.apache.org/docs/format/FlightSql.html) protocol, which is natively supported by Spice. This enables any JDBC-compatible application to connect to Spice, execute queries, and retrieve data.
```
Comment (Jeadie) [inline spiceaidocs/docs/api/jdbc/index.md]:
```suggestion
Spice supports JDBC clients through a JDBC driver implementation based on the [Flight SQL](https://arrow.apache.org/docs/format/FlightSql.html) protocol. This enables any JDBC-compatible application to connect to Spice, execute queries, and retrieve data.
```

File: spiceaidocs/docs/api/jdbc/index.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+[Full instruction](/clients/tableau)
+
+**JetBrains DataGrip**:
+
+- In Database Explorer menu, select "+" and choose "Driver"
+- Follow the steps to add the JDBC `.jar` file
+
+[Full instruction](/clients/jetbrains-datagrip)
+
+**DBeave**:
```
Comment (phillipleblanc) [inline spiceaidocs/docs/api/jdbc/index.md]:
```suggestion
**DBeaver**:
```

File: spiceaidocs/docs/api/jdbc/index.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+[JDBC](https://docs.oracle.com/javase/tutorial/jdbc/basics/index.html) (Java Database Connectivity) is a standard API for connecting to and interacting with databases.
+
+Spice supports JDBC clients through a JDBC driver implementation based on the [Flight SQL](https://arrow.apache.org/docs/format/FlightSql.html) protocol, which is natively supported by Spice. This enables any JDBC-compatible application to connect to Spice, execute queries, and retrieve data.
+
+## Download and install the Flight SQL JDBC driver
+
+### Download the Flight SQL JDBC driver
+
+- Visit the [Flight SQL JDBC driver](https://central.sonatype.com/artifact/org.apache.arrow/flight-sql-jdbc-driver/) page
```
Comment (Jeadie) [inline spiceaidocs/docs/api/jdbc/index.md]:
```suggestion
- Find the appropriate [Flight SQL JDBC driver](https://central.sonatype.com/artifact/org.apache.arrow/flight-sql-jdbc-driver/versions) version. 
```

File: spiceaidocs/docs/api/jdbc/index.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+- Select the **Versions** tab
```
Comment (Jeadie) [inline spiceaidocs/docs/api/jdbc/index.md]:
```suggestion
```

File: spiceaidocs/docs/api/jdbc/index.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+- In the DBeaver application menu bar, open the "Database" menu and choose: "Driver Manager"
+- Click the "New" button and follow instructions to add JDBC `.jar` file.
+
+[Full instruction](/clients/DBeaver)
+
+## Configure JDBC connection
+
+1. Use the following configuration settings:
+
+- **URL**: `jdbc:arrow-flight-sql://{host}:{port}?useEncryption=false&disableCertificateVerification=true`
```
Comment (phillipleblanc) [inline spiceaidocs/docs/api/jdbc/index.md]:
```suggestion
- **URL**: `jdbc:arrow-flight-sql://{host}:{port}`
```

Comment (phillipleblanc) [reply spiceaidocs/docs/api/jdbc/index.md]: We should add a note saying that if you aren't using Spice with TLS enabled, you can use `?useEncryption=false&disableCertificateVerification=true` - but it shouldn't be the default we recommend.

File: spiceaidocs/docs/api/jdbc/index.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+- In the DBeaver application menu bar, open the "Database" menu and choose: "Driver Manager"
+- Click the "New" button and follow instructions to add JDBC `.jar` file.
+
+[Full instruction](/clients/DBeaver)
+
+## Configure JDBC connection
+
+1. Use the following configuration settings:
+
+- **URL**: `jdbc:arrow-flight-sql://{host}:{port}`
```
Comment (phillipleblanc) [inline spiceaidocs/docs/api/jdbc/index.md:51]: We should add a note saying that if you aren't using Spice with TLS enabled (with link to TLS docs), you can use `?useEncryption=false&disableCertificateVerification=true` - but it shouldn't be the default we recommend.

**Review by Sevenannn — DISMISSED** [review]

**Review by Jeadie — APPROVED** [review]

**Pull request merged**